python/backend/backend/discover/algorithms/deci.py

```diff
@@ -0,0 +1,321 @@
+        adj_mat = deci_model.get_adj_matrix_tensor(
+            do_round=False, samples=1, most_likely_graph=True
+        )
+        adj_mat = adj_mat.squeeze(0)
+        adj_mat = adj_mat.detach().cpu().numpy().astype(np.float64)
+        ate_mat = DeciRunner._compute_deci_average_treatment_effect(
+            deci_model, azua_dataset
+        )
+
+        deci_model.eval()
+        numCols = self._prepared_data.shape[1]
+        intervention_mask = torch.cat(
+            (
+                torch.zeros(math.ceil(numCols / 2), dtype=torch.bool),
+                torch.ones(math.floor(numCols / 2), dtype=torch.bool),
+            ),
+            0,
+        )
+        intervention_values = torch.rand(math.floor(numCols / 2))
+        gumbel_max_regions = torch.LongTensor(
+            deci_model.variables.processed_cols_by_type["categorical"]
+        )
+        gt_zero_region = torch.LongTensor(
+            [
+                j
+                for i in deci_model.variables.processed_cols_by_type["binary"]
+                for j in i
+            ]
+        )
+
+        deci_inference_inputs = (
+            torch.rand([1, numCols]),
+            torch.from_numpy(adj_mat).float(),
+            intervention_mask,
+            intervention_values,
+            gumbel_max_regions,
+            gt_zero_region,
+        )
+        input_names = [
+            "X",
+            "W_adj",
+            "intervention_mask",
+            "intervention_values",
+            "gumbel_max_regions",
+            "gt_zero_region",
+        ]
+        output_names = ["inference_results"]
+        onnx_output = io.BytesIO()
+        torch.onnx.export(
+            torch.jit.script(deci_model.ICGNN),
+            deci_inference_inputs,
+            onnx_output,
+            input_names=input_names,
+            output_names=output_names,
+            # opset_version=17, # TODO: Once pytorch implements opset 17 we can use nn.LayerNorm
+            opset_version=11,
+            export_params=True,
+            # verbose=True,
+            # keep_initializers_as_inputs=True,
+            dynamic_axes={
+                # 'intervention_mask' : {0 : 'intervention_count'},    # variable length axes
+                "intervention_values": {0: "intervention_count"}
+            },
+        )
+        onnx_base64 = base64.b64encode(onnx_output.getvalue())
+
+        deci_graph = networkx.convert_matrix.from_numpy_matrix(
+            adj_mat, create_using=networkx.DiGraph
+        )
+        deci_ate_graph = networkx.convert_matrix.from_numpy_matrix(
+            ate_mat, create_using=networkx.DiGraph
+        )
+
+        for n1, n2, d in deci_graph.edges(data=True):
+            # TODO: validate what to do when there is no edge in the ate graph
+            #       currently returning weight 0
+            ate = deci_ate_graph.get_edge_data(n1, n2, default={"weight": 0})["weight"]
```

Comment:
`default={"weight": 0}` fixes an internal server error when the edge does not exist in the ate graph